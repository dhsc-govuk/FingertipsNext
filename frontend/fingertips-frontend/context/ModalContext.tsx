import {
  createContext,
  FC,
  ReactNode,
  useContext,
  useMemo,
  useState,
} from 'react';
import { ModalPane } from '@/components/molecules/ModalPane/ModalPane';

export interface Modal {
  content?: ReactNode;
  title?: string;
}

export const ModalContext = createContext<{
  modal: Modal;
  setModal: (newPreview: Modal) => void;
}>({ modal: {}, setModal: () => undefined });

export const useModal = () => {
  const { modal, setModal } = useContext(ModalContext);
  const { content, title } = modal;
  return { modal, setModal, content, title };
};

interface ModalProviderProps {
  children: ReactNode;
}

export const ModalProvider: FC<ModalProviderProps> = ({ children }) => {
  const [modal, setModal] = useState<Modal>({});
  const providerValue = useMemo(
    () => ({
      modal,
      setModal,
    }),
    [modal]
  );

  return (
    <ModalContext.Provider value={providerValue}>
      {children}
      <ModalPane />
    </ModalContext.Provider>
  );
};
