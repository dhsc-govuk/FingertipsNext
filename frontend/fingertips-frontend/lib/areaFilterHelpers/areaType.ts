import { AreaType } from '@/generated-sources/ft-api-client';

export const combinedAuthoritiesAreaType = {
  key: 'combined-authorities',
  name: 'Combined Authorities',
  hierarchyName: 'Admin',
  level: 3,
};

export const countiesAndUnitaryAuthoritiesAreaType = {
  key: 'counties-and-unitary-authorities',
  name: 'Counties and Unitary Authorities',
  hierarchyName: 'Admin',
  level: 4,
};

export const districtAndUnitaryAuthoritiesAreaType = {
  key: 'districts-and-unitary-authorities',
  name: 'Districts and Unitary Authorities',
  hierarchyName: 'Admin',
  level: 5,
};

export const gpsAreaType = {
  key: 'gps',
  name: 'GPs',
  hierarchyName: 'NHS',
  level: 6,
};

export const nhsIntegratedCareBoardsAreaType = {
  key: 'nhs-integrated-care-boards',
  name: 'NHS Integrated Care Boards',
  hierarchyName: 'NHS',
  level: 3,
};

export const nhsPrimaryCareNetworksAreaType = {
  key: 'nhs-primary-care-networks',
  name: 'NHS Primary Care Networks',
  hierarchyName: 'NHS',
  level: 5,
};

export const nhsRegionsAreaType = {
  key: 'nhs-regions',
  name: 'NHS Regions',
  hierarchyName: 'NHS',
  level: 2,
};

export const nhsSubIntegratedCareBoardsAreaType = {
  key: 'nhs-sub-integrated-care-boards',
  name: 'NHS Sub Integrated Care Boards',
  hierarchyName: 'NHS',
  level: 4,
};

export const regionsAreaType = {
  key: 'regions',
  name: 'Regions',
  hierarchyName: 'Admin',
  level: 2,
};

export const englandAreaType = {
  key: 'england',
  name: 'England',
  hierarchyName: 'All',
  level: 1,
};

export const adminHierarchyAreaTypes: AreaType[] = [
  combinedAuthoritiesAreaType,
  countiesAndUnitaryAuthoritiesAreaType,
  districtAndUnitaryAuthoritiesAreaType,
  regionsAreaType,
];

export const nhsHierarchyAreaTypes: AreaType[] = [
  gpsAreaType,
  nhsIntegratedCareBoardsAreaType,
  nhsPrimaryCareNetworksAreaType,
  nhsRegionsAreaType,
  nhsSubIntegratedCareBoardsAreaType,
];

const allHierarchyAreaTypes: AreaType[] = [englandAreaType];

export const allAreaTypes = [
  ...adminHierarchyAreaTypes,
  ...nhsHierarchyAreaTypes,
  ...allHierarchyAreaTypes,
];

export const adminHierarchyAreaTypeKeys = [
  'combined-authorities',
  'counties-and-unitary-authorities',
  'districts-and-unitary-authorities',
  'regions',
] as const;

export const nhsHierarchyAreaTypeKeys = [
  'gps',
  'nhs-integrated-care-boards',
  'nhs-primary-care-networks',
  'nhs-regions',
  'nhs-sub-integrated-care-boards',
] as const;

export const allHierarchyAreaTypesKey = ['england'] as const;

const _areaTypeKeys = [
  ...adminHierarchyAreaTypeKeys,
  ...nhsHierarchyAreaTypeKeys,
  ...allHierarchyAreaTypesKey,
] as const;

export type AreaTypeKeys = (typeof _areaTypeKeys)[number];
