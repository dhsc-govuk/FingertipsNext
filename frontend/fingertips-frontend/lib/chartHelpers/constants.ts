export const areaCodeForEngland = 'E92000001';
export const englandAreaString = 'England';
export const indicatorIdForPopulation = 337;
