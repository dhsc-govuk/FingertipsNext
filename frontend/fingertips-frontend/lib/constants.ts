export const siteTitle = 'Find public health data';

export const siteDescription = `${siteTitle} is a rich source of indicators across a range of health and wellbeing themes designed to support JSNA and commissioning to improve health and wellbeing, and reduce inequalities.`;

export const spaceSeparatedPattern = /^\d+(\s+\d+)+$/;
