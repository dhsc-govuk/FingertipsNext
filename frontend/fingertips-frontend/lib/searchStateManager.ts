import { asArray } from './pageHelpers';

export enum SearchParams {
  SearchedIndicator = 'si',
  IndicatorsSelected = 'is',
  AreasSelected = 'as',
  AreaTypeSelected = 'ats',
  GroupTypeSelected = 'gts',
  ConfidenceIntervalSelected = 'cis',
}

export type SearchParamKeys = `${SearchParams}`;

const multiValueParams = [
  SearchParams.IndicatorsSelected as string,
  SearchParams.AreasSelected as string,
  SearchParams.ConfidenceIntervalSelected as string,
];

export type SearchStateParams = {
  [SearchParams.SearchedIndicator]?: string;
  [SearchParams.IndicatorsSelected]?: string[];
  [SearchParams.AreasSelected]?: string[];
  [SearchParams.AreaTypeSelected]?: string;
  [SearchParams.GroupTypeSelected]?: string;
  [SearchParams.ConfidenceIntervalSelected]?: string[];
};

const isMultiValueTypeParam = (searchParamKey: SearchParamKeys) =>
  multiValueParams.includes(searchParamKey);

export class SearchStateManager {
  private searchState: SearchStateParams;
  private searchStateParams: URLSearchParams;

  constructor(searchState: SearchStateParams = {}) {
    this.searchState = {};
    this.searchStateParams = new URLSearchParams();

    this.searchState = searchState;
  }

  private constructPath(path: string) {
    if (this.searchStateParams.size === 0) {
      return path;
    }
    return `${path}?${this.searchStateParams.toString()}`;
  }

  private addParamToPath(searchParamKey: SearchParamKeys) {
    const paramsToAdd = asArray(this.searchState[searchParamKey]);

    if (paramsToAdd.length > 0) {
      paramsToAdd.forEach((paramToAdd) => {
        this.searchStateParams.append(searchParamKey, paramToAdd);
      });
    }
  }

  public addParamValueToState(
    searchParamKey: SearchParamKeys,
    paramToAdd: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = this.searchState[searchParamKey];

      const currentParamStateAsArray = asArray(currentParamState);
      const doesParamExist = currentParamStateAsArray.find(
        (currentParam) => currentParam === paramToAdd
      );

      if (!doesParamExist) {
        const newState: SearchStateParams = {
          ...this.searchState,
          [searchParamKey]: [...currentParamStateAsArray, paramToAdd],
        };

        this.searchState = newState;
      }
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: paramToAdd,
      };

      this.searchState = newState;
    }
  }

  public removeParamValueFromState(
    searchParamKey: SearchParamKeys,
    paramToRemove?: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = asArray(this.searchState[searchParamKey]);

      const newParamState = currentParamState?.filter((paramValue) => {
        return paramValue !== paramToRemove;
      });

      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: newParamState,
      };

      this.searchState = newState;
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: undefined,
      };

      this.searchState = newState;
    }
  }

  public removeAllParamFromState(searchParamKey: SearchParamKeys) {
    this.searchState[searchParamKey] = undefined;
  }

  public static setStateFromParams(params: URLSearchParams) {
    const newState = Object.values(SearchParams).reduce<SearchStateParams>(
      (state, searchParamKey) => {
        if (isMultiValueTypeParam(searchParamKey)) {
          const paramValues = params.getAll(searchParamKey);

          if (paramValues.length > 0) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValues,
            };

            return newState;
          }
        } else {
          const paramValue = params.get(searchParamKey);

          if (paramValue) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValue,
            };

            return newState;
          }
        }
        return state;
      },
      {}
    );

    const searchStateManager = new SearchStateManager(newState);
    return searchStateManager;
  }

  public static setStateFromSearchStateParams(params?: SearchStateParams) {
    if (params) {
      const newState = Object.values(SearchParams).reduce<SearchStateParams>(
        (state, searchParamKey) => {
          if (isMultiValueTypeParam(searchParamKey)) {
            const paramValues = asArray(params[searchParamKey]);

            if (paramValues.length > 0) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValues,
              };

              return newState;
            }
          } else {
            const paramValue = params[searchParamKey];

            if (paramValue) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValue,
              };

              return newState;
            }
          }
          return state;
        },
        {}
      );

      const searchStateManager = new SearchStateManager(newState);
      return searchStateManager;
    }
    return new SearchStateManager();
  }

  public getSearchState() {
    return this.searchState;
  }

  public generatePath(path: string) {
    this.searchStateParams = new URLSearchParams();

    Object.values(SearchParams).forEach((searchParamKey) => {
      this.addParamToPath(searchParamKey);
    });

    return this.constructPath(path);
  }
}
