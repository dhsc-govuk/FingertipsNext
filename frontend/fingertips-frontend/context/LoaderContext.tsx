'use client';

import React, {
  ReactNode,
  createContext,
  useContext,
  useMemo,
  useState,
  useEffect,
  Suspense,
  Dispatch,
  SetStateAction,
} from 'react';
import { usePathname } from 'next/navigation';
import { LoadingBox } from 'govuk-react';
import { useSearchStateParams } from '@/components/hooks/useSearchStateParams';
import { SearchStateParams } from '@/lib/searchStateManager';

export type LoaderContext = {
  getIsLoading: () => boolean;
  setIsLoading: (loading: boolean) => void;
};

type LoaderContextProvider = {
  children: ReactNode;
};

const LoaderContext = createContext<LoaderContext | undefined>(undefined);

export const LoaderProvider: React.FC<LoaderContextProvider> = ({
  children,
}) => {
  const pathname = usePathname();
  const [searchState, setSearchState] = useState<SearchStateParams>({});

  const [isLoading, setIsLoading] = useState<boolean>(true);

  const contextValue: LoaderContext = useMemo(
    () => ({
      getIsLoading: () => isLoading,
      setIsLoading,
    }),
    [isLoading, setIsLoading]
  );

  useEffect(() => {
    setIsLoading(false);
  }, [setIsLoading, pathname, searchState]);

  return (
    <LoaderContext.Provider value={contextValue}>
      <Suspense>
        <SuspendedUseSearchStateParams setSearchState={setSearchState} />
      </Suspense>
      <LoadingBox loading={isLoading} timeIn={200} timeOut={200}>
        {children}
      </LoadingBox>
    </LoaderContext.Provider>
  );
};

export const useLoadingState = () => {
  const context = useContext(LoaderContext);

  if (!context) {
    throw new Error('useLoadingState must be used within a LoaderProvider');
  }

  return context;
};

// Next has a very peculiar quirk where useSearchState cannot be called server
// side, which works fine during dev but build will fail.
// Here we're creating a component which will not render server side by
// suspending it until client mounts at which point it can set the loading
// context value, all this is more complicated because loader context is in a
// layout.
function SuspendedUseSearchStateParams({
  setSearchState,
}: Readonly<{ setSearchState: Dispatch<SetStateAction<SearchStateParams>> }>) {
  const searchParams = useSearchStateParams();
  useEffect(() => {
    setSearchState(searchParams);
  }, [searchParams, setSearchState]);

  return null;
}
