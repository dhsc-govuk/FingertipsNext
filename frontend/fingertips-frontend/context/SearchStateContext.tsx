'use client';

import React, { ReactNode, createContext, useContext, useState } from 'react';
import { SearchStateParams } from '@/lib/searchStateManager';

export type SearchStateContext = {
  getSearchState: () => SearchStateParams;
  setSearchState: (state: SearchStateParams) => void;
};

type SearchStateContextProvider = {
  children: ReactNode;
};

const SearchStateContext = createContext<SearchStateContext | undefined>(
  undefined
);

export const SearchStateProvider: React.FC<SearchStateContextProvider> = ({
  children,
}) => {
  const [searchState, setSearchState] = useState<SearchStateParams>({});

  const contextValue: SearchStateContext = {
    getSearchState: () => searchState,
    setSearchState,
  };

  return (
    <SearchStateContext.Provider value={contextValue}>
      {children}
    </SearchStateContext.Provider>
  );
};

export const useSearchState = () => {
  const context = useContext(SearchStateContext);

  if (!context) {
    throw new Error('useLoadingState must be used within a LoaderProvider');
  }

  return context;
};
