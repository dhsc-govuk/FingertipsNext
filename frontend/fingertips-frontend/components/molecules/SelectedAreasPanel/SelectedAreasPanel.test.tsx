import { mockAreaDataForNHSRegion } from '@/mock/data/areaData';
import { render, screen, within } from '@testing-library/react';
import { SelectedAreasPanel } from '.';
import { SearchParams } from '@/lib/searchStateManager';
import userEvent from '@testing-library/user-event';
import { nhsPrimaryCareNetworksAreaType } from '@/lib/areaFilterHelpers/areaType';
import { ALL_AREAS_SELECTED } from '@/lib/areaFilterHelpers/constants';
import { LoaderContext } from '@/context/LoaderContext';

const mockSelectedAreasData = [
  mockAreaDataForNHSRegion['E40000007'],
  mockAreaDataForNHSRegion['E40000012'],
];

const mockPath = 'some-mock-path';
const mockReplace = jest.fn();

jest.mock('next/navigation', () => {
  const originalModule = jest.requireActual('next/navigation');

  return {
    ...originalModule,
    usePathname: () => mockPath,
    useSearchParams: () => {},
    useRouter: jest.fn().mockImplementation(() => ({
      replace: mockReplace,
    })),
  };
});

const mockSetIsLoading = jest.fn();
const mockLoaderContext: LoaderContext = {
  getIsLoading: jest.fn(),
  setIsLoading: mockSetIsLoading,
};

jest.mock('@/context/LoaderContext', () => {
  return {
    useLoadingState: () => mockLoaderContext,
  };
});

describe('SelectedAreasPanel', () => {
  describe('When there is a group area selected', () => {
    it('snapshot test', () => {
      const container = render(
        <SelectedAreasPanel
          areaFilterData={{
            availableAreas: mockSelectedAreasData,
          }}
          searchState={{
            [SearchParams.GroupAreaSelected]: ALL_AREAS_SELECTED,
            [SearchParams.AreaTypeSelected]: nhsPrimaryCareNetworksAreaType.key,
          }}
        />
      );

      expect(container.asFragment()).toMatchSnapshot();
    });

    it('should render the group area selected panel and not render the standard selected areas panel', () => {
      render(
        <SelectedAreasPanel
          areaFilterData={{
            availableAreas: mockSelectedAreasData,
          }}
          searchState={{
            [SearchParams.GroupAreaSelected]: ALL_AREAS_SELECTED,
            [SearchParams.AreaTypeSelected]: nhsPrimaryCareNetworksAreaType.key,
          }}
        />
      );

      expect(
        screen.getByTestId('group-selected-areas-panel')
      ).toBeInTheDocument();
      expect(
        screen.queryByTestId('standard-selected-areas-panel')
      ).not.toBeInTheDocument();
    });

    it('should render the selected areas count and single pill for the group area selected', () => {
      render(
        <SelectedAreasPanel
          areaFilterData={{
            availableAreas: mockSelectedAreasData,
          }}
          searchState={{
            [SearchParams.GroupAreaSelected]: ALL_AREAS_SELECTED,
            [SearchParams.AreaTypeSelected]: nhsPrimaryCareNetworksAreaType.key,
          }}
        />
      );

      expect(screen.getByText(/Selected areas \(2\)/i)).toBeInTheDocument();
      expect(screen.getAllByTestId('pill-container')).toHaveLength(1);
    });

    it('should remove the group area selected from the url when the remove icon is clicked', async () => {
      const expectedPath = [
        `${mockPath}`,
        `?${SearchParams.AreaTypeSelected}=${nhsPrimaryCareNetworksAreaType.key}`,
      ].join('');

      const user = userEvent.setup();
      render(
        <SelectedAreasPanel
          areaFilterData={{
            availableAreas: mockSelectedAreasData,
          }}
          searchState={{
            [SearchParams.GroupAreaSelected]: ALL_AREAS_SELECTED,
            [SearchParams.AreaTypeSelected]: nhsPrimaryCareNetworksAreaType.key,
          }}
        />
      );

      const firstSelectedAreaPill = screen.getAllByTestId('pill-container')[0];
      await user.click(
        within(firstSelectedAreaPill).getByTestId('remove-icon-div')
      );

      expect(mockReplace).toHaveBeenCalledWith(expectedPath, {
        scroll: false,
      });
    });

    it('should call setIsLoading with true when an area type is selected', async () => {
      const user = userEvent.setup();
      render(
        <SelectedAreasPanel
          areaFilterData={{
            availableAreas: mockSelectedAreasData,
          }}
          searchState={{
            [SearchParams.GroupAreaSelected]: ALL_AREAS_SELECTED,
            [SearchParams.AreaTypeSelected]: nhsPrimaryCareNetworksAreaType.key,
          }}
        />
      );

      const firstSelectedAreaPill = screen.getAllByTestId('pill-container')[0];
      await user.click(
        within(firstSelectedAreaPill).getByTestId('remove-icon-div')
      );

      expect(mockSetIsLoading).toHaveBeenCalledWith(true);
    });
  });

  describe('When there are areas selected', () => {
    it('snapshot test', () => {
      const container = render(
        <SelectedAreasPanel selectedAreasData={mockSelectedAreasData} />
      );

      expect(container.asFragment()).toMatchSnapshot();
    });

    it('should render the standard area selected panel and not render the group selected areas panel', () => {
      render(<SelectedAreasPanel />);

      expect(
        screen.queryByTestId('group-selected-areas-panel')
      ).not.toBeInTheDocument();
      expect(
        screen.getByTestId('standard-selected-areas-panel')
      ).toBeInTheDocument();
    });

    it('should not render the selected areas pill when there are no areas selected', () => {
      render(<SelectedAreasPanel />);

      expect(screen.getByText(/Selected areas \(0\)/i)).toBeInTheDocument();
      expect(screen.queryByTestId('pill-container')).not.toBeInTheDocument();
    });

    it('should render the selected areas when there are areas selected', () => {
      render(<SelectedAreasPanel selectedAreasData={mockSelectedAreasData} />);

      expect(screen.getByText(/Selected areas \(2\)/i)).toBeInTheDocument();
      expect(screen.getAllByTestId('pill-container')).toHaveLength(2);
    });

    it('should remove the area selected from the url when the remove icon is clicked for the area selected', async () => {
      const expectedPath = [
        `${mockPath}`,
        `?${SearchParams.AreasSelected}=E40000012`,
        `&${SearchParams.AreaTypeSelected}=NHS+Regions`,
      ].join('');

      const user = userEvent.setup();
      render(
        <SelectedAreasPanel
          selectedAreasData={mockSelectedAreasData}
          searchState={{
            [SearchParams.AreasSelected]: ['E40000012', 'E40000007'],
            [SearchParams.AreaTypeSelected]: 'NHS Regions',
          }}
        />
      );

      const firstSelectedAreaPill = screen.getAllByTestId('pill-container')[0];
      await user.click(
        within(firstSelectedAreaPill).getByTestId('remove-icon-div')
      );

      expect(mockReplace).toHaveBeenCalledWith(expectedPath, {
        scroll: false,
      });
    });

    it('should call setIsLoading with true when an area type is selected', async () => {
      const user = userEvent.setup();
      render(
        <SelectedAreasPanel
          selectedAreasData={mockSelectedAreasData}
          searchState={{
            [SearchParams.AreasSelected]: ['E40000012', 'E40000007'],
            [SearchParams.AreaTypeSelected]: 'NHS Regions',
          }}
        />
      );

      const firstSelectedAreaPill = screen.getAllByTestId('pill-container')[0];
      await user.click(
        within(firstSelectedAreaPill).getByTestId('remove-icon-div')
      );

      expect(mockSetIsLoading).toHaveBeenCalledWith(true);
    });
  });
});
