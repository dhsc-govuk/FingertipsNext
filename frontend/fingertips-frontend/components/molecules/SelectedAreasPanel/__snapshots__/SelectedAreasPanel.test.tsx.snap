// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedAreasPanel When there are areas selected snapshot test 1`] = `
<DocumentFragment>
  .c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c5 {
  cursor: pointer;
}

.c3 {
  background-color: white;
  border: 1px #D1D2D3 solid;
  border-radius: 5px;
  padding: 0.5em 0.3125em 0.5em 0em;
  max-width: 100%;
  margin: 0.3125em 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  word-wrap: break-word;
  padding: 0em 1em 0em 0.75em;
  font-size: 16px;
  margin-bottom: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
}

.c4:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c7 {
  font-weight: 500;
}

.c8 {
  color: #505A5F;
}

.c0 {
  padding-bottom: 1.5em;
}

.c2 {
  font-weight: bold;
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c6 {
    font-family: sans-serif;
  }
}

<div
    class="c0"
    data-testid="selected-areas-panel"
  >
    <div
      data-testid="standard-selected-areas-panel"
    >
      <span
        class="c1 c2"
      >
        Selected areas (2)
      </span>
      <div
        class="c3"
        data-testid="pill-container"
      >
        <button
          class="c4"
          data-testid="remove-icon-div"
        >
          <svg
            class="c5"
            data-testid="x-icon"
            fill="none"
            height="12"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            viewBox="0 0 25 25"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="6"
              y2="18"
            />
            <line
              x1="6"
              x2="18"
              y1="6"
              y2="18"
            />
          </svg>
        </button>
        <div
          class="c6"
          data-testid="filter-name"
        >
          <p
            style="margin: 0px;"
          >
            <span
              class="c7"
            >
              East of England NHS Region
            </span>
             
            <span
              class="c8"
            >
              NHS Regions
            </span>
          </p>
        </div>
      </div>
      <div
        class="c3"
        data-testid="pill-container"
      >
        <button
          class="c4"
          data-testid="remove-icon-div"
        >
          <svg
            class="c5"
            data-testid="x-icon"
            fill="none"
            height="12"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            viewBox="0 0 25 25"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="6"
              y2="18"
            />
            <line
              x1="6"
              x2="18"
              y1="6"
              y2="18"
            />
          </svg>
        </button>
        <div
          class="c6"
          data-testid="filter-name"
        >
          <p
            style="margin: 0px;"
          >
            <span
              class="c7"
            >
              North East and Yorkshire NHS Region
            </span>
             
            <span
              class="c8"
            >
              NHS Regions
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectedAreasPanel When there is a group area selected snapshot test 1`] = `
<DocumentFragment>
  .c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c5 {
  cursor: pointer;
}

.c3 {
  background-color: white;
  border: 1px #D1D2D3 solid;
  border-radius: 5px;
  padding: 0.5em 0.3125em 0.5em 0em;
  max-width: 100%;
  margin: 0.3125em 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  word-wrap: break-word;
  padding: 0em 1em 0em 0.75em;
  font-size: 16px;
  margin-bottom: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
}

.c4:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c7 {
  color: #505A5F;
}

.c0 {
  padding-bottom: 1.5em;
}

.c2 {
  font-weight: bold;
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c6 {
    font-family: sans-serif;
  }
}

<div
    class="c0"
    data-testid="selected-areas-panel"
  >
    <div
      data-testid="group-selected-areas-panel"
    >
      <span
        class="c1 c2"
      >
        Selected areas (2)
      </span>
      <div
        class="c3"
        data-testid="pill-container"
      >
        <button
          class="c4"
          data-testid="remove-icon-div"
        >
          <svg
            class="c5"
            data-testid="x-icon"
            fill="none"
            height="12"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            viewBox="0 0 25 25"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="6"
              y2="18"
            />
            <line
              x1="6"
              x2="18"
              y1="6"
              y2="18"
            />
          </svg>
        </button>
        <div
          class="c6"
          data-testid="filter-name"
        >
          <p
            style="margin: 0px;"
          >
            All areas in  
            <span
              class="c7"
            >
              NHS Primary Care Networks
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
